tests/unit/utils/test_serialize_dataclass.py

```diff
@@ -0,0 +1,212 @@
+    ):
+        """Test the conversion of the dataclass to a dict.
+
+        :param content_view: The content view
+        :param serialization_tuple: The suffix, serialization format
+        """
+        content = ContentTestSimple()
+        result = content.asdict(
+            serialization_format=serialization_tuple[1],
+            content_view=content_view,
+        )
+        assert result == {"attr_01": False, "attr_02": 2, "attr_03": "three"}
+
+    @staticmethod
+    def test_content_to_json(content_view: ContentView):
+        """Test the conversion of the dataclass to json.
+
+        :param content_view: The content view
+        """
+        content = ContentTestSimple()
+        result = serialize(
+            content=content,
+            content_view=content_view,
+            serialization_format=SerializationFormat.JSON,
+        )
+        expected = '{\n    "attr_01": false,\n    "attr_02": 2,\n    "attr_03": "three"\n}'
+        assert result == expected
+
+    @staticmethod
+    def test_content_to_yaml(content_view: ContentView):
+        """Test the conversion of the dataclass to yaml.
+
+        :param content_view: The content view
+        """
+        content = ContentTestSimple()
+        result = serialize(
+            content=content,
+            content_view=content_view,
+            serialization_format=SerializationFormat.YAML,
+        )
+        expected = "---\nattr_01: false\nattr_02: 2\nattr_03: three\n"
+        assert result == expected
+
+    @staticmethod
+    @pytest.mark.parametrize(**ParametrizeFormat()._asdict())
+    def test_content_to_dict_override(
+        content_view: ContentView,
+        serialization_tuple: Tuple[str, SerializationFormat],
+    ):
+        """Test the conversion of the dataclass with overrides to a ``dict``.
+
+        :param content_view: The content view
+        :param serialization_tuple: The suffix, serialization format
+        """
+        content = ContentTestOverride()
+        result = content.asdict(
+            content_view=content_view,
+            serialization_format=serialization_tuple[1],
+        )
+        for _key, value in result.items():
+            assert value.endswith(f"_{serialization_tuple[0]}_{str(content_view)}")
+
+    @staticmethod
+    def test_content_to_json_override(content_view: ContentView):
+        """Test the conversion of the dataclass with overrides to ``JSON``.
+
+        :param content_view: The content view
+        """
+        content = ContentTestOverride()
+        result = serialize(
+            content=content,
+            content_view=content_view,
+            serialization_format=SerializationFormat.JSON,
+        )
+        result_template = (
+            '{{\n    "attr_01": "False__j_{view}",\n'
+            '    "attr_02": "2__j_{view}",\n'
+            '    "attr_03": "three__j_{view}"\n}}'
+        )
+        assert result == result_template.format(view=content_view)
+
+    @staticmethod
```

Comment:
This is extremely weird and unnatural in the pytest ecosystem. It makes no sense to nest the tests under a class. Just make pure functions instead.

Reply:
[@pytest.mark.parametrize](https://docs.pytest.org/en/latest/how-to/parametrize.html#pytest-mark-parametrize) allows one to define multiple sets of arguments and fixtures at the test function or class.

Seems documented: https://docs.pytest.org/en/latest/how-to/parametrize.html

Reply:
It does. But the whole "put pure functions under a class" part is pointless.

Reply:
ok, it was just convenient :)

Reply:
done